include/nlohmann/json.hpp

```diff
@@ -3486,31 +3489,32 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     access by reference
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.0.0
+    @since version 1.0.0; template added in version 3.10.0
 
     @liveexample{The example below shows how object elements can be read and
     written using `at()`. It also demonstrates the different exceptions that
     can be thrown.,at__object_t_key_type}
     */
-    reference at(const typename object_t::key_type& key)
+    template < class KeyT, typename std::enable_if <
+                   !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value&& (
+#if defined(JSON_HAS_CPP_17)
+                       std::is_same<typename std::decay<KeyT>::type, std::string_view>::value ||
+#endif
+                       std::is_convertible<KeyT, typename object_t::key_type>::value), int >::type    = 0 >
+    reference at(KeyT && key)
     {
         // at only works for objects
-        if (JSON_HEDLEY_LIKELY(is_object()))
+        if (JSON_HEDLEY_UNLIKELY(!is_object()))
         {
-            JSON_TRY
-            {
-                return set_parent(m_value.object->at(key));
-            }
-            JSON_CATCH (std::out_of_range&)
-            {
-                // create better exception explanation
-                JSON_THROW(out_of_range::create(403, "key '" + key + "' not found", *this));
-            }
+            JSON_THROW(type_error::create(304, "cannot use at() with " + std::string(type_name()), *this));
         }
-        else
+
+        auto it = m_value.object->find(std::forward<KeyT>(key));
+        if (it == m_value.object->end())
         {
-            JSON_THROW(type_error::create(304, "cannot use at() with " + std::string(type_name()), *this));
+            JSON_THROW(out_of_range::create(403, "key '" + std::string(key) + "' not found", *this));
```

Comment:
Using key here after it was std::forward'd in the find() statement makes the assumption that std::map::find() will never modify the passed-in key under any circumstances.

Better would probably be to make this function take a const& to KeyT, instead of an &&, since you *don't* want the passed in key to be modified, even if the caller of your function doesn't care, you do.

Reply:
Oh dear. Totally forgot about that. I'll change it and see if I missed other places as well.